### DISCUSSION TITLE
Is custom compiling still needed for MQTT TLS?

### DISCUSSION THREAD

**Discussion by EpicLPer** [arendst/Tasmota#18530] [Q&A] [answered]
Heya!

I'm currently working on a remote project that requires MQTT over Internet, after some research I came to the conclusion that I have to compile a custom version of Tasmota with MQTT TLS and fingerprint learning enabled to get it connected to HiveMQ. I tried ESPHome before but it doesn't have support for MQTT TLS included for the ESP8266.

Is all of the below still needed to actually achieve my goal? Or is support for this built into a firmware version of Tasmota already and I just haven't stumbled across this yet? I tried it before with the standard firmware but wasn't able to connect to HiveMQ since they require TLS.

Thanks!

```
// Enable MQTT TLS for HiveMQ

#ifndef USE_MQTT_TLS 

#define USE_MQTT_TLS                             // Use TLS for MQTT connection (+34.5k code, +7.0k mem and +4.8k additional during connection handshake)
#define MQTT_TLS_ENABLED       true              // [SetOption103] Enable TLS mode (requires TLS version)
#define MQTT_PORT              8883              // [MqttPort] MQTT port (10123 on CloudMQTT)
#define MQTT_FINGERPRINT1      0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00
#endif
```

**Comment by sfromis** [comment, marked as answer]
Yes, standard builds for ESP8266 does not include TLS. Not likely to change as this feature pushes available RAM and processor resources. However, there is an "unoffical" build including TLS in https://github.com/tasmota/install/tree/firmware/firmware/unofficial (not available right now).

If you select a device based on ESP32, standard Tasmota builds does include TLS.

**Reply by EpicLPer** [reply]
Ah, thanks for the link! Didn't know there were "unofficial" builds too, or I guess rather unsupported config ones(?).
I could use a ESP32 worst case, but would have to redesign and reprint my custom 3D enclosure cause I didn't plan for the fact that TLS might not be supported, tbh was one of the last things I would've ever thought of ^^"

Thanks tho! Means I can use this unofficial build at least :)

**Reply by Jason2866** [reply]
All repos online again.

**Reply by Jason2866** [reply]
The TLS is as stable as an official version. It is not provided as official version since TLS eats up a lot of precious IRAM which is missing for other usefull features.

**Reply by hnaeem07** [reply]
Hello, I want built my TLS enabled tasmota bin file without webserver (//#define USE_WEBSERVER) , i cannot enter the TLSKeys1 and TLSKeyss2 on the serial interface, plus mqtt is not connected since its TLS enabled. can hardcode the Key files ?

Really appreciate if anyone could help.

Thank you in advance.

**Comment by pkkrusty** [comment]
I did some recent testing with TLS, and was doing ok on RAM with only a few drivers and extras defined. But like Jason said, will need to keep your eye on it. #18339

**Reply by sfromis** [reply]
Beware that Jason was not talking about RAM (heap) but about IRAM.
